cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -888,11 +888,5 @@ mod test {
             size_of::<PrimaryMap<VCodeConstant, VCodeConstantData>>(),
             24
         );
-        assert_eq!(size_of::<HashMap<Constant, VCodeConstant>>(), 48);
-        assert_eq!(size_of::<HashMap<*const [u8], VCodeConstant>>(), 48);
-        assert_eq!(size_of::<VCodeConstants>(), 120);
-        assert_eq!(size_of_val(&VCodeConstants::with_capacity(0)), 120);
-        // TODO This structure could use some significant memory-size optimization. The use of
-        // HashMap to deduplicate both pool and well-known constants is clearly an issue.
```

Comment:
Here's a suggestion for documentation to replace this:
```
// TODO The VCodeConstants structure's memory size could be further optimized. 
// With certain versions of Rust, each `HashMap` in `VCodeConstants` occupied at
// least 48 bytes, making an empty `VCodeConstants` cost 120 bytes.
```

Reply:
Good idea, thanks!